tikv/raftstore/raftlog/custom.go

```diff
@@ -0,0 +1,244 @@
+package raftlog
+
+import (
+	"encoding/binary"
+	"fmt"
+	"unsafe"
+
+	"github.com/pingcap/kvproto/pkg/raft_cmdpb"
+)
+
+type CustomRaftLogType byte
+
+const (
+	CustomRaftLogFlag byte = 64
+
+	TypePrewrite            CustomRaftLogType = 1
+	TypeCommit              CustomRaftLogType = 2
+	TypeRolback             CustomRaftLogType = 3
+	TypePessimisticLock     CustomRaftLogType = 4
+	TypePessimisticRollback CustomRaftLogType = 5
+)
+
+// CustomRaftLog is the raft log format for unistore to store Prewrite/Commit/PessimisticLock.
+//  | flag(1) | type(1) | version(2) | header(40) | entries
+//
+// It reduces the cost of marshal/unmarshal and avoid DB lookup during apply.
+type CustomRaftLog struct {
```

Comment:
Should we add some bench cases for the log generation and serialization

Reply:
It is all plain append, the bench result is not very helpful.